server.go

```diff
@@ -2128,7 +2128,7 @@ func (s *server) InboundPeerConnected(conn net.Conn) {
 		// we'll close out this connection s.t there's only a single
 		// connection between us.
 		localPub := s.identityPriv.PubKey()
-		if !shouldDropLocalConnection(localPub, nodePub) {
+		if !connectedPeer.inbound && !shouldDropLocalConnection(localPub, nodePub) {
```

Comment:
Comments here and below no longer match the rationale behind the logic.

Reply:
I changed the comments to be more specific and to contain description for the additional logic.